images/linux/scripts/helpers/etc-environment.sh

```diff
@@ -0,0 +1,64 @@
+#!/bin/bash
+################################################################################
+##  File:  etc-environment.sh
+##  Desc:  Helper functions for source and modify /etc/environment
+################################################################################
+
+# NB: sed expression use '%' as a delimiter in order to simplify handling
+#     values containg slashes (i.e. directory path)
+#     The values containing '%' will break the functions
+
+function getEtcEnvironmentVar {
+    var_name="$1"
+    # remove `var_name=` and possible quotes from the line
+    grep "^${var_name}=" /etc/environment |sed -E "s%^${var_name}=\"?([^\"]+)\"?.*$%\1%"
+}
+
+function addEtcEnvironmentVar {
+    var_name="$1"
+    var_value="$2"
+
+    echo "$var_name=\"$var_value\"" | sudo tee -a /etc/environment
+}
+
+function replaceEtcEnvironmentVar {
+    var_name="$1"
+    var_value="$2"
+
+    # modify /etc/environemnt in place by replacing a string that begins with var_name
+    sudo sed -ie "s%^${var_name}=.*$%${var_name}=\"${var_value}\"%" /etc/environment
+}
```

Comment:
It looks like these functions `getEtcEnvironmentVar`, `addEtcEnvironmentVar`, `replaceEtcEnvironmentVar` are internal and not intendant to use in other scripts.
As I know, bash doesn't support local function but may be we can use specific naming for them? like `_getEtcEnvironmentVar`, `_addEtcEnvironmentVar`, `_replaceEtcEnvironmentVar`? Or left comment in this file

Reply:
I do not think so. While currently these functions are not used outside but they are not intended to be private and some day the may be useful in outer code. Nothing should prevent developers from using them if they needed